ignite/engine/__init__.py

```diff
@@ -41,6 +47,208 @@ def _prepare_batch(
     )
 
 
+def _check_arg(on_tpu: bool, amp_mode: Optional[str], scaler: Optional["torch.cuda.amp.GradScaler"]) -> Optional[str]:
```

Comment:
Move before `create_supervised_trainer` ?

Reply:
For readability?
I defined after `_prepare_batch` just for placing internal functions in one place.

Reply:
This function is defined line 50 but used line 250. The idea is to reduce the gap for readability, yes.

Reply:
Should `_prepare_batch` move too?